crt_portal/cts_forms/views.py

```diff
@@ -336,6 +336,103 @@ def serialize_data(report, request, report_id):
+            profile_form.save()
+            messages.add_message(request, messages.SUCCESS, 'Successfully Saved Profile')
+
+            """redirects back to /form/view but all filter params are not perserved. """
+            return redirect('/form/view')
+        else:
+            """Write Errors into messages and return invalid profile_form back to IndexView"""
+            for key in profile_form.errors:
+                errors = '; '.join(profile_form.errors[key])
+                error_msg = f'Could not save profile: {errors}'
+                messages.add_message(request, messages.ERROR, error_msg)
+
+            # Placeholder to represent requested Get params.
+            report_query, query_filters = report_filter(request.GET)
+
+            # Sort data based on request from params, default to `created_date` of complaint
+            sort = ['-create_date']
+            per_page = 15
+            page = 1
+
+            requested_reports = report_query.order_by(*sort)
+            paginator = Paginator(requested_reports, per_page)
+            requested_reports, page_format = pagination(paginator, page, per_page)
+
+            sort_state = {}
+            # make sure the links for this page have the same paging, sorting, filtering etc.
+            page_args = f'?per_page={per_page}'
+
+            # process filter query params
+            filter_args = ''
+            for query_item in query_filters.keys():
+                arg = query_item
+                for item in query_filters[query_item]:
+                    filter_args = filter_args + f'&{arg}={item}'
+            page_args += filter_args
+
+            # process sort query params
+            sort_args = ''
+            for sort_item in sort:
+                if sort_item[0] == SORT_DESC_CHAR:
+                    sort_state.update({sort_item[1::]: True})
+                else:
+                    sort_state.update({sort_item: False})
+
+                sort_args += f'&sort={sort_item}'
+            page_args += sort_args
+
+            all_args_encoded = urllib.parse.quote(f'{page_args}&page={page}')
+
+            data = []
+
+            paginated_offset = page_format['page_range_start'] - 1
+            for index, report in enumerate(requested_reports):
+                p_class_list = format_protected_class(
+                    report.protected_class.all().order_by('form_order'),
+                    report.other_class,
+                )
+                if report.other_class:
+                    p_class_list.append(report.other_class)
+                if len(p_class_list) > 3:
+                    p_class_list = p_class_list[:3]
+                    p_class_list[2] = f'{p_class_list[2]}...'
+
+                data.append({
+                    "report": report,
+                    "report_protected_classes": p_class_list,
+                    "url": f'{report.id}?next={all_args_encoded}&index={paginated_offset + index}',
+                })
+
```

Comment:
This looks to be a copy/paste from the current `IndexView` implementation. Where we need functionality that exists elsewhere in the code we should decompose and pull into a function that can be called in multiple places. Please do that here.